nemo_aligner/models/nlp/gpt/megatron_gpt_knowledge_distillation.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import hydra
+import torch
+from megatron.core import parallel_state, tensor_parallel
+from megatron.core.num_microbatches_calculator import get_micro_batch_size, get_num_microbatches
+from megatron.core.parallel_state import get_tensor_model_parallel_group
+from megatron.core.pipeline_parallel.schedules import get_forward_backward_func
+from omegaconf.dictconfig import DictConfig
+from pytorch_lightning.trainer.trainer import Trainer
+
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_model import MegatronGPTModel
+from nemo.collections.nlp.modules.common.megatron.utils import (
+    average_losses_across_data_parallel_group,
+    get_iterator_k_split,
+)
+from nemo.collections.nlp.parts.mixins.nlp_adapter_mixins import NLPAdapterModelMixin
+from nemo.collections.nlp.parts.utils_funcs import get_last_rank
+from nemo_aligner.models.alignable_interface import SupervisedInterface
+from nemo_aligner.utils.distributed import _TopKLogitsCrossEntropy
+from nemo_aligner.utils.train_utils import (
+    finish_validation_step,
+    grad_reductions,
+    prepare_for_training_step,
+    prepare_for_validation_step,
+    set_sync_funcs,
+)
+
+
+class GPTKnowledgeDistillationModel(NLPAdapterModelMixin, MegatronGPTModel, SupervisedInterface):
+    def __init__(self, cfg: DictConfig, trainer: Trainer):
+        super().__init__(cfg, trainer=trainer)
+
+        self.target_logits_scale = self.cfg.knowledge_distillation.get("target_logits_scale", 1.0)
+        self.logits_scale = self.cfg.knowledge_distillation.get("logits_scale", 1.0)
+
+        self.kd_loss = self.cfg.knowledge_distillation.get("kd_loss", "bwd_kl")
+        self.jsd_weight = self.cfg.knowledge_distillation.get("jsd_weight", 0.5)
```

Comment:
`jsd_weight` is useless now since we don't support `kd_loss=jsd`. Can you please remove it?